src/flashbar/collapsible-flashbar.tsx

```diff
@@ -93,6 +93,10 @@ export default function CollapsibleFlashbar({ items, ...restProps }: FlashbarPro
         focusFlashById(ref.current, mostRecentItem.id);
       }
     }
+    // When collapsing, scroll up if necessary to avoid losing track of the focused button
+    else if (wasFlashbarStackExpanded && !isFlashbarStackExpanded && notificationBarRef.current) {
+      scrollElementIntoView(notificationBarRef.current);
```

Comment:
By moving this here, this will be called regardless of motion being enabled or not. Before, it was under a condition `if (initialAnimationState)`, and `initialAnimationState` is only initialized if motion is enabled.

Reply:
1. Let's not use dnd-kit code in unrelated use-cases. We have our own version of it by the way: https://github.com/cloudscape-design/components/blob/2a073a233c1fd0704d55cb043269fab4d2a90819/src/internal/hooks/use-previous/index.ts#L8
2. I think, this code does not need this value at all, because the hook already depends on `isFlashbarStackExpanded`, so if `isFlashbarStackExpanded === false` it automatically means it was `true` before. If you want to skip initial render, there is `useEffectOnUpdate` which is more idiomatic approach to solve this: https://github.com/cloudscape-design/components/blob/2a073a233c1fd0704d55cb043269fab4d2a90819/src/internal/hooks/use-effect-on-update.ts#L6

Reply:
Oh, imported from the wrong place, thanks for spotting this.

Ideally I'd like to keep using a layout effect instead of a "normal" effect to prevent flickering, i.e, a frame where the flashbar collapsed but the scrolling hasn't happened yet.

Reply:
Fixed the import

Reply:
Because `useEffectOnUpdate` uses `useEffect` and not `useLayoutEffect`, the scroll is delayed to the next render. See test failure: https://github.com/cloudscape-design/components/actions/runs/6576775598/job/17866925600

Reply:
`findNotificationBar().click()` uses vanilla `HTMLElement.click` function instead of operating via our test utils.

Use our test utils and I expect it will work fine